### PR TITLE
Add leader peer for get region response.

### PR DESCRIPTION
**Pull request by qiuyesuifeng**
@disksing  @siddontang @shenli  PTAL

### CODE REVIEW

File: server/balancer_worker_test.go [hunk 1]
```diff
@@ -39,8 +39,9 @@ func (s *testBalancerWorkerSuite) TestBalancerWorker(c *C) {
 	clusterInfo := s.ts.newClusterInfo(c)
 	c.Assert(clusterInfo, NotNil)
 
-	region := clusterInfo.regions.getRegion([]byte("a"))
+	region, leaderPeer := clusterInfo.regions.getRegionAndLeader([]byte("a"))
```
Comment (ngaut) [inline server/balancer_worker_test.go]: s/leaderPeer/leader/

File: server/cache_test.go [hunk 2]
```diff
@@ -94,8 +94,8 @@ func (s *testClusterCacheSuite) TestCache(c *C) {
 
 	// There is only one region now, directly use it for test.
 	regionKey := []byte("a")
-	region, err := cluster.getRegion(regionKey)
-	c.Assert(err, IsNil)
+	region, leaderPeer := cluster.getRegionAndLeader(regionKey)
```
Comment (ngaut) [inline server/cache_test.go]: s/leaderPeer/leader/

File: server/cluster.go [hunk 3]
```diff
@@ -327,13 +327,13 @@ func (c *RaftCluster) cacheAllRegions() error {
 	return nil
 }
 
-func (c *RaftCluster) getRegion(regionKey []byte) (*metapb.Region, error) {
-	return c.cachedCluster.regions.getRegion(regionKey), nil
+func (c *RaftCluster) getRegionAndLeader(regionKey []byte) (*metapb.Region, *metapb.Peer) {
```
Comment (ngaut) [inline server/cluster.go]: s/getRegionAndLeader/regionAndLeader/

File: server/api/region.go [hunk 4]
```diff
@@ -64,7 +64,7 @@ func (h *regionHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	region, leader := cluster.GetRegionAndLeader(regionID)
+	region, leader := cluster.GetRegionAndLeaderByID(regionID)
```
Comment (siddontang) [inline server/api/region.go]: why not using GetRegion directly?

**Comment by shenli** [comment]
LGTM

File: server/api/region.go [hunk 5]
```diff
@@ -64,7 +64,7 @@ func (h *regionHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	region, leader := cluster.GetRegionAndLeader(regionID)
+	region, leader := cluster.GetRegionByID(regionID)
```
Comment (siddontang) [inline server/api/region.go]: do we need using a structure to hold region and leader?
If we want to return another info, how do we do?

Comment (qiuyesuifeng) [inline server/api/region.go]: If that, we should warp with a new struct, but now i think it is OK.

File: server/cache.go [hunk 6]
```diff
@@ -170,25 +170,31 @@ func newRegionsInfo() *regionsInfo {
...
-		return cloneRegion(region)
+	if !keyInRegion(regionKey, region) {
+		return nil, nil
 	}
 
-	return nil
+	regionID := region.GetId()
+	leaderStoreID, ok := r.leaders.regionStores[regionID]
+	if ok {
+		return cloneRegion(region), leaderPeer(region, leaderStoreID)
```
Comment (disksing) [inline server/cache.go]: I think you need to pass the cloned region to `leaderPeer()`.
```
cloned := cloneRegion(reigon)
return cloned, leaderPeer(cloned, leaderStoreID)
```

Comment (qiuyesuifeng) [inline server/cache.go]: Seems not different, we have already used a read lock to protect it.

**Comment by qiuyesuifeng** [comment]
PTAL

**Comment by siddontang** [comment]
LGTM

**Pull request merged**